### ISSUE TITLE
Make Differs and Patchers package public

### ISSUE THREAD

**Issue by Benestar** [wmde/WikibaseDataModelServices#127]
To be able to use those components outside of this package for other entity differs they must be package public. <s>I wonder what would be a nice namespace name...</s>

**Comment by thiemowmde** [comment]
Please collect a list of the specific classes that should become public. Do we really need all of them? Is there an other way?

**Comment by Benestar** [comment]
We could also create interfaces and offer a factory but that seems to be an unnecessary overhead.

The classes needed for a MediaInfoDiffer and MediaInfoPatcher are (afaik) `StatementListDiffer`, `StatementListPatcher` and when merged `TermListPatcher`.

**Comment by JeroenDeDauw** [comment]
> I wonder what would be a nice namespace name...

Something wrong with the one they are currently in? It already has differs, patchers and diffs in it.

**Comment by Benestar** [comment]
I rather referred to `\Internal` which suggests that the classes are package private.

**Comment by JeroenDeDauw** [comment]
Yeah I get that. So if they are no longer package private, they go out of the Internal sub-NS, which leaves the in the Diff NS. Anything wrong with that NS? (That was my question)

**Comment by Benestar** [comment]
Ah, sorry. Nothing wrong with that namespace besides it gets a bit crowded. But that's perhaps not a problem yet.

**Comment by Benestar** [comment]
Btw, do we only want to move the classes listed above out of the Internal namespace or move everything out of it? It might not be obvious why `TermListPatcher` is public while `AliasGroupListPatcher` is private.

**Comment by JeroenDeDauw** [comment]
To me it seems strange to just move some of them out of these, if they are all doing equivalent things.
